React/Modules/RCTAsyncLocalStorage.m

```diff
@@ -403,34 +362,22 @@ - (NSDictionary *)_writeEntry:(NSArray<NSString *> *)entry changedManifest:(BOOL
     return;
   }
   NSMutableArray<NSDictionary *> *errors;
-  BOOL changedManifest = NO;
   for (NSString *key in keys) {
     NSDictionary *keyError = RCTErrorForKey(key);
     if (!keyError) {
-      if (_manifest[key] == (id)kCFNull) {
-        NSString *filePath = [self _filePathForKey:key];
-        [[NSFileManager defaultManager] removeItemAtPath:filePath error:nil];
-        [RCTGetCache() removeObjectForKey:key];
-        // remove the key from manifest, but no need to mark as changed just for
-        // this, as the cost of checking again next time is negligible.
-        [_manifest removeObjectForKey:key];
-      } else if (_manifest[key]) {
-        changedManifest = YES;
-        [_manifest removeObjectForKey:key];
-      }
+      NSString *filePath = [self _filePathForKey:key];
```

Comment:
A lot of changes to the logic here - why get rid of the changedManifest bool, etc?